services/web/server/src/simcore_service_webserver/workspaces/_trash_handlers.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from aiohttp import web
+from servicelib.aiohttp import status
+from servicelib.aiohttp.requests_validation import (
+    parse_request_path_parameters_as,
+    parse_request_query_parameters_as,
+)
+
+from .._meta import API_VTAG as VTAG
+from ..application_settings_utils import requires_dev_feature_enabled
+from ..login.decorators import get_user_id, login_required
+from ..products.api import get_product_name
+from ..security.decorators import permission_required
+from . import _trash_api
+from ._exceptions_handlers import handle_plugin_requests_exceptions
+from ._models import WorkspacesPathParams, WorkspaceTrashQueryParams
+
+_logger = logging.getLogger(__name__)
+
+
+routes = web.RouteTableDef()
+
+
+@routes.post(f"/{VTAG}/workspaces/{{workspace_id}}:trash", name="trash_workspace")
+@requires_dev_feature_enabled
+@login_required
+@permission_required("workspaces.*")
+@handle_plugin_requests_exceptions
+async def trash_workspace(request: web.Request):
```

Comment:
the number of decorators becomes quite impressive ;)

Reply:
all good here ;-)